pkg/crm/crm_appinst.go

```diff
@@ -68,6 +68,14 @@ func (s *CRMData) appInstChanged(ctx context.Context, old *edgeproto.AppInst, ne
 				new.Fields = edgeproto.AppInstAllFields
 			} else {
 				new.Fields = old.GetDiffFields(new).Fields()
+				// Special case for dns update - only possible if appinst exists
+				fmap := edgeproto.MakeFieldMap(new.Fields)
+				if !fmap.Has(edgeproto.AppInstFieldState) {
+					if fmap.Has(edgeproto.AppInstFieldUri) {
+						_ = s.AppInstDNSChanged(ctx, s.cloudletKey, old, new, responseSender)
+						return
+					}
+				}
```

Comment:
I was planning to add a separate DNS Change API for the CCRM, since this is so specialized and we might not want to go through the same logic as a generalized update. Would you rather use the existing update api from CCRM?
You other comment about Adding Annotation for the previous dns name seem to suggest you'd prefer reusing existing update api to CCRM. Am I getting it right?

Reply:
Moved the logic to controller-data.go.